### ISSUE TITLE
Usage of `Task.project` prevents use of configuration cache

### ISSUE THREAD

**Issue by bmuschko** [kobylynskyi/graphql-java-codegen#1151]
## Issue Description

Run build with `--configuration-cache`. You will receive the following error message for your plugin.

> invocation of `Task.project` at execution time is unsupported.

## Steps to Reproduce

Apply your plugin. Run build with `--configuration-cache`.

## Expected Result

Gradle does not report configuration cache error anymore.

## Actual Result

Error message is rendered.

## Your Environment and Setup

* **graphql-java-codegen version**: *[v5.7.1](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v5.7.1)*
* **Build tool**: *Gradle 8.0*

**Comment by bmuschko** [comment]
Is this something you are planning to fix soon or would you rather want to receive a pull request?

**Comment by kobylynskyi** [comment]
Yes, you are right. I've removed it from 5.7.2 release.

**Comment by esfomeado** [comment]
@kobylynskyi I gave it a try but I wasn't able to get it to work.

Are you able to look at this?

Thanks

**Comment by vvvlasov** [comment]
Hi! This is still an issue as of 5.10.0, do you need any help to get it done?

**Comment by esfomeado** [comment]
@vvvlasov Yes please

**Comment by xenoterracide** [comment]
yep, went to look/report today and this blocks me from using this plugin. I guess I'll go about this the otherway code -> schema instead of schema -> code
